[Tan Cheong Hsien Ryan] iP

ketamethane opened the pull request:
### Welcome to Duke

> “Sooner or later, those who win are those who think they can.” — Paul Tournier

Duke helps you keep track of your tasks! It is

- text-based
- quick to master
- ~~quick~~ efficient to use

All you need to do is to...
1. download it from [here](https://github.com/ketamethane/ip/blob/master/out/artifacts/iP_jar/iP.jar)
2. click on **View Raw**
3. add your tasks
4. let it manage your tasks for you 😊 

It is **FREE**! :)

Features:

- [x] Managing tasks
- [ ] managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ketamethane committed:
Add Level-1

ketamethane committed:
Add level-2

ketamethane committed:
Add Level-3

ketamethane committed:
Add Level-4

ketamethane committed:
Add level-5 and level-6

ketamethane committed:
Add A-TextUiTesting

ketamethane committed:
Add Level-7

ketamethane committed:
Add Level-8

ketamethane committed:
Merge branch 'branch-Level-7'

ketamethane committed:
Merge branch 'branch-Level-8'

# Conflicts: implementation of LocalDate attribute for deadline affects helper functions in Duke.java
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

ketamethane committed:
Add A-MoreOOP and A-Packages

ketamethane committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

ketamethane committed:
Add A-Gradle

ketamethane committed:
Add A-JUnit and rename package

ketamethane committed:
Add A-JavaDoc

ketamethane committed:
Add A-CodingStandard

ketamethane committed:
Add Level-9

ketamethane committed:
Merge branch 'branch-Level-9'

ketamethane committed:
Merge branch 'branch-A-JavaDoc' and add JavaDoc for classes

# Conflicts: Comment Overlap
#	src/main/java/seedu/duke/TaskList.java

ketamethane committed:
Add A-Jar

pwjj2000 commented on src/main/java/seedu/duke/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+    protected LocalDate dueDate;
+    protected String dueTime;
+
+    /**\
+     * Creates a Deadline task that has a date but no time included.
+     * @param description description of Deadline.
```
Great and consistent work on the Javadoc documentation throughout your submission! 

Although, it might be better to leave a blank line between the description and the `@param` so that it is more aligned to the convention. Similar comments for the rest of the documentation as well.

Please note that you might have accidentally included an additional backslash `\` in line 14 as well.

pwjj2000 commented on src/main/java/seedu/duke/DukeException.java:
```diff
@@ -0,0 +1,16 @@
+
+
+
+//    public static void main(String[] args) {
+//        try {
+//
+//        }
+//    }
```
Will it be more reader-friendly for other reviewers if commented out code is removed instead? Similar questions for other parts of the code

pwjj2000 commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+//        String command = inputArr[0];
+        System.out.println("____________________________________________________");
+
+        switch (command) {
```
Very neat work by using switch cases! 👍

pwjj2000 commented on src/main/java/seedu/duke/Storage.java:
```diff
@@ -0,0 +1,194 @@
+     */
+    public void addDeadlineOrEvent(ArrayList<Task> taskList, String s, String isMarked, String description, String datetime){
+//        System.out.println("type:" + s);
+        if (s.equals("D")){
+//            System.out.println("added D");
+            String[] dateAndTime = datetime.split(" ");
+            System.out.println(dateAndTime.length);
+            if (dateAndTime.length == 2) {
+                String dueDate = dateAndTime[0];
+                String dueTime = dateAndTime[1];
+                LocalDate localDate = getLocalDate(dueDate);
+                Task deadline = new Deadline(description, localDate, dueTime);
+                if (isMarked.equals("1")) {
+                    deadline.mark();
+                }
+                taskList.add(deadline);
+            } else {
+                String dueDate = dateAndTime[0];
+                LocalDate localDate = getLocalDate(dueDate);
+                Task deadline = new Deadline(description, localDate);
+                if (isMarked.equals("1")) {
+                    deadline.mark();
+                }
+                taskList.add(deadline);
+            }
+        } else {
+//            System.out.println("added E");
+            Event event = new Event(description, datetime);
+            if (isMarked.equals("1")) {
+                event.mark();
+            }
+            taskList.add(event);
+        }
```
Have you considered using switch cases as well here since the conditionals are a little deeply nested?

pwjj2000 commented on src/main/java/seedu/duke/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+                System.out.println("  " + todo);
+                break;
+            case "deadline":
+                String[] removeTaskType2 = input.split("deadline ");
+                String desAndBy = String.join("", removeTaskType2);
+                String[] sliceByDesAndBy = desAndBy.split(" /by ");
+                String description2 = sliceByDesAndBy[0];
+                String dueDateAndTime = sliceByDesAndBy[1];
+                String[] dateAndTime = dueDateAndTime.split(" ");
+                System.out.println(dateAndTime.length);
+                if (dateAndTime.length == 2) {
+                    String dueDate = dateAndTime[0];
+                    String dueTime = dateAndTime[1];
+                    LocalDate localDate = Storage.getLocalDate(dueDate);
+                    Task deadline = new Deadline(description2, localDate, dueTime);
+                    taskList.add(deadline);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + deadline);
+                } else {
+                    String dueDate = dateAndTime[0];
+                    LocalDate localDate = Storage.getLocalDate(dueDate);
+                    Task deadline = new Deadline(description2, localDate);
+                    taskList.add(deadline);
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + deadline);
+                }
+                break;
+            case "event":
+                String[] removeTaskType3 = input.split("event ");
+                String desAndBy2 = String.join("", removeTaskType3);
+                String[] sliceByDesAndBy2 = desAndBy2.split(" /at ");
+                String description3 = sliceByDesAndBy2[0];
+                String dueTime2 = sliceByDesAndBy2[1];
+                Task event = new Event(description3, dueTime2);
+                taskList.add(event);
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + event);
+                break;
+
+        }
```
Could you double check on the indentation for this again? If I remember correctly, `case` should be aligned with `switch`, though I could be wrong as well.

pwjj2000 reviewed (commented):
Great piece of code with no major issues!

ketamethane committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts: wrapping of text
#	src/main/java/seedu/duke/Parser.java
#	src/main/java/seedu/duke/Storage.java
#	src/main/java/seedu/duke/Task.java
#	src/main/java/seedu/duke/Ui.java

ketamethane committed:
Add Level-10

ketamethane committed:
Merge branch 'Level-10'

ketamethane committed:
Update Jar file

ketamethane committed:
Add A-Assertions

ketamethane committed:
Add A-CodeQuality

ketamethane committed:
Add A-CodeQuality

Currently, there is a lot of dead code (e.g. print statements)

Remove dead code

Currently, addTask() in TaskList.java is a bit clutter, with 2 levels of
abstraction.

Extract functions within the switch cases to 3 separate functions

ketamethane committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/ketamethane/ip into branch-A-CodeQuality

ketamethane committed:
Merge pull request #2 from ketamethane/branch-A-Assertions

Add A-Assertions

ketamethane committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/seedu/duke/Storage.java

ketamethane committed:
Merge branch 'master' into branch-A-Assertions

ketamethane committed:
Merge pull request #1 from ketamethane/branch-A-CodeQuality

Add A-CodeQuality

ketamethane committed:
Merge branch 'master' into branch-A-Assertions

ketamethane committed:
Merge branch 'master' into branch-A-CodeQuality

ketamethane committed:
Add BCD-Extension

ketamethane committed:
Merge branch 'branch-A-CodeQuality'

ketamethane committed:
Merge branch 'branch-A-Assertions'

ketamethane committed:
Update README.md

ketamethane committed:
Update README.md

ketamethane committed:
Mass update

If data directory is not in jar file, it is created.
Bug: tasks.txt folder added inside data folder, which is redundant

Remove Duke2 which is a dead code
Add Ui.png
Remove dead print statements

ketamethane committed:
Merge branch 'master' of https://github.com/ketamethane/ip

ketamethane committed:
Update README.md

ketamethane committed:
Add checkstyleMain to build.gradle

ketamethane committed:
Remove I/O tests

ketamethane committed:
Refactor of config folder

ketamethane committed:
Improve checkstyle

ketamethane committed:
Delete .github/workflows directory

ketamethane committed:
Remove CI file

ketamethane committed:
Merge branch 'master' of https://github.com/ketamethane/ip

ketamethane committed:
Update README.md

ketamethane committed:
Add files via upload

ketamethane committed:
Update README.md

ketamethane committed:
Delete Ui.png.jpg

ketamethane committed:
Change image path to relative

ketamethane committed:
Credit adapted lines

ketamethane committed:
Update README.md

ketamethane committed:
Remove resizing window option

ketamethane committed:
Merge branch 'master' of https://github.com/ketamethane/ip

ketamethane committed:
Fix minor grammar mistake in README

ketamethane committed:
Converted deku.png to jpg and back to png format

ketamethane committed:
Merge branch 'master' of https://github.com/ketamethane/ip
